File: Monitor-Shop-FrontEnd/src/app/common/CartDetail.ts
import { Cart } from "./cart";
import { Product } from "./Product";

export class CartDetail {
    'id': number;
    'quantity': number;
    'price': number;
    'product': Product;
    'cart': Cart;

    constructor(id:number, quantity:number, price:number, product:Product, cart: Cart) {
        this.id = id;
        this.quantity = quantity;
        this.price = price;
        this.product = product;
        this.cart = cart;
    }
}

File: Monitor-Shop-BackEnd/src/main/java/com/mshop/entity/CartDetail.java
package com.mshop.entity;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@SuppressWarnings("serial")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "cart_details")
public class CartDetail implements Serializable{
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private int quantity;
	private double price;
	
	@ManyToOne
	@JoinColumn(name = "product_id")
	private Product product;
	
	@ManyToOne
	@JoinColumn(name = "cart_id")
	private Cart cart;
}
